transcom/mymove
# MB-7648 filter list by requested move date

## Description

This PR filters the list by the requested pickup date (AKA the requested move date).  It also fixes a bug in the UI that displayed the incorrect date in requested pickup date column. The day shown is one day prior to the actual pickup date. There is an on going conversation about this bug: https://ustcdp3.slack.com/archives/CP497TGAU/p1618418221039300

## Reviewer Notes

In my test, how do I properly assert the requested pickup date? Currently, there is an assertion in there and it is not working.  When the shipment is created, it is created with the default requested shipment date instead of the asserted date. 

## Setup

`make server_test`

To verify in the UI complete the following steps:
- start the server: `make server_run`
- start the client: `make client_run`
- navigate to `http://officelocal:3000/counseling/queue` as the TOO
- in the requested pickup date, filter the column
![Screen Shot 2021-04-14 at 10 22 37 AM](https://user-images.githubusercontent.com/56279459/114760189-88c8e380-9d0b-11eb-9a68-27c77a5c0b94.png)



## Code Review Verification Steps

* Any new migrations/schema changes:

## References

* [Jira story](tbd) for this change
* [this article](tbd) explains more about the approach used.

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._
